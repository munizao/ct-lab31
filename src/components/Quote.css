h1 {
  text-align: center;
}

.MainBox {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  border: .2rem solid black;
  padding: 0px;
  margin: 0px 20vw 0px 20vw;
  ul {
    width: 20vw;
    margin: 0px;
    padding: 0px;
    background-color: #0d0852;
    li {
      margin: 0px;
      padding: .5em;
      list-style-type: none;
      color: white;
      &:hover {
        background-color: #382fb6;
      }
      &.Selected {
        background-color: #2fb658;
      }
    }    
  }
  .Quote {
    width: 40vw;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 1rem;
  }
  img {
    height: 50vmin;
  }
}
