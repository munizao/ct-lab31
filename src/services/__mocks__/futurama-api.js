export const quote = jest.fn(() => Promise.resolve({
  character:	'Professor Farnsworth',
  quote:	'I\'m as spry as a 140 year old. See? I only broke one ankle.',
  image:	'https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/farnsworth.png'
}));
