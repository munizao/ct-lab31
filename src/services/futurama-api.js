export const quoteAPI = (characterName, numQuotes) => {
  if(characterName) {
    return fetch(`https://futuramaapi.herokuapp.com/api/characters/${characterName.split(' ').join('-')}/${numQuotes}`)
      .then(res => res.json());
  }
  else {
    return fetch(`https://futuramaapi.herokuapp.com/api/quotes/${numQuotes}`)
      .then(res => res.json());
  }
};
export const charactersAPI = (page) => {
  return fetch(`https://futuramaapi.herokuapp.com/api/v2/characters?page=${page}`)
    .then(res => res.json()) // The keys on this API are all capitalized, which is stupid.
    .then(characters => {
      return characters.map(character => Object.fromEntries(Object.entries(character)
        .map(entry => [entry[0].toLowerCase(), entry[1]]))); 
    });
};
