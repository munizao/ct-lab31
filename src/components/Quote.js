import React from 'react';
import { quote } from '../services/futurama-api';

export default class Quote extends React.Component {
  state = {
    character: '',
    quote: '',
    image: ''
  };

  handleClick = () => {
    quote()
      .then((res) => {
        this.setState(res);
      });
  }

  render() {
    return (
      <div>
        <img src={this.state.image} /><p>{this.state.quote}</p><p>{this.state.character}</p>
        <button onClick={this.handleClick}>Get new Quote</button>
      </div>
    );
  }
}
