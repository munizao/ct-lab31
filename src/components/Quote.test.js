import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote.js';
import { quote } from '../services/futurama-api.js';
jest.mock('../services/futurama-api.js');

describe('Quote component', () => {
  it('renders Quote', () => {
    const wrapper = shallow(<Quote />);
    expect(wrapper).toMatchSnapshot();
  });

  it('fetches from the api', () => {
    const wrapper = shallow(<Quote />);
    const quoteFetcher = wrapper.instance();
    quoteFetcher.fetch()
      .then(() => {
        expect(wrapper.state('character')).toEqual('Professor Farnsworth');
        expect(wrapper.state('quote')).toEqual('I\'m as spry as a 140 year old. See? I only broke one ankle.');
        expect(wrapper.state('image')).toEqual('https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/farnsworth.png');
      });
    expect(quote).toHaveBeenCalled();
  });

});

