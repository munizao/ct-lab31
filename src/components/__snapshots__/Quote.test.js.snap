// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders Quote 1`] = `
<div>
  <img
    src=""
  />
  <p />
  <p />
  <button
    onClick={[Function]}
  >
    Get new Quote
  </button>
</div>
`;
